internal/pkg/agent/install/flavors.go

```diff
@@ -118,7 +118,7 @@ func ApplyFlavor(versionedHome string, flavor FlavorDefinition) error {
 	}
 
 	for _, ftr := range filesToRemove {
-		if removeErr := os.RemoveAll(ftr); !os.IsNotExist(removeErr) {
+		if removeErr := os.RemoveAll(ftr); removeErr != nil && !os.IsNotExist(removeErr) {
```

Comment:
Same as above.

What bug is this fixing? It's not clear there's a test for it.

Reply:
this is path that was not hit before. in case removeErr is nil it will still go inside this condition